dart/dynamics/SoftBodyNode.h

```diff
@@ -361,7 +356,7 @@ class SoftBodyNode : public BodyNode
   UniqueProperties mSoftP;
 
   /// \brief Soft mesh shape belonging to this node.
-  std::shared_ptr<SoftMeshShape> mSoftShape;
+  ShapeNode* mSoftShapeNode;
```

Comment:
I think we should use a ``NodePtr<ShapeNode>`` here instead to make sure it doesn't get deleted due to the user fiddling around with the ShapeNodes. I also think we need a more robust way of handling Soft Shapes, especially when it comes to things like cloning, but we can defer that issue for now.